recipes/pistache/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.29.1"
+
+
+class PistacheConan(ConanFile):
+    name = "pistache"
+    license = "Apache-2.0"
+    homepage = "https://github.com/pistacheio/pistache"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("http", "rest", "framework", "networking")
+    description = "Pistache is a modern and elegant HTTP and REST framework for C++"
+    settings = "os", "compiler", "arch", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False], "with_ssl": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "with_ssl": True}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compilers = {
+            "gcc": "7",
+            "clang": "6",
+        }
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Pistache is only support by Linux.")
+
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 17)
+        minimum_compiler = compilers.get(str(self.settings.compiler))
+        if minimum_compiler:
+            if tools.Version(self.settings.compiler.version) < minimum_compiler:
+                raise ConanInvalidConfiguration("Pistache requires c++17, which your compiler does not support.")
+        else:
+            self.output.warn("Pistache requires c++17, but this compiler is unknown to this recipe. Assuming your compiler supports c++17.")
+
+    def requirements(self):
+        self.requires("rapidjson/1.1.0")
+        if self.options.with_ssl:
+            self.requires("openssl/1.1.1h")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob("pistache-*")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_sources(self):
```

Comment:
upstream forces PIC: https://github.com/pistacheio/pistache/blob/ae146fa773019118ae48ef48fa91ba8888de79aa/src/CMakeLists.txt#L15
You have to remove this line, and set CMAKE_POSITION_INDEPENDENT_CODE in build() based on fPIC option value (because upstream relies on OBJECT target).

Reply:
Removed, done.